Description: Optionally skip glance search tests
 python-elasticsearch not in Ubuntu main; skip tests
 if not installed.
Author: James Page <james.page@ubuntu.com>
Forwarded: not-needed

--- a/glance/tests/unit/test_search.py
+++ b/glance/tests/unit/test_search.py
@@ -16,11 +16,17 @@
 import datetime
 
 import mock
+import testtools
 
 from oslo_utils import timeutils
 
-from glance.search.plugins import images as images_plugin
-from glance.search.plugins import metadefs as metadefs_plugin
+try:
+    from glance.search.plugins import images as images_plugin
+    from glance.search.plugins import metadefs as metadefs_plugin
+    elasticsearch = True
+except ImportError:
+    elasticsearch = False
+
 import glance.tests.unit.utils as unit_test_utils
 import glance.tests.utils as test_utils
 
@@ -165,6 +171,7 @@ def _db_tag_fixture(name, **kwargs):
 
 
 class TestImageLoaderPlugin(test_utils.BaseTestCase):
+    @testtools.skipIf(not elasticsearch, "elasticsearch not installed")
     def setUp(self):
         super(TestImageLoaderPlugin, self).setUp()
         self.db = unit_test_utils.FakeDB()
@@ -394,6 +401,7 @@ class TestImageLoaderPlugin(test_utils.B
 
 
 class TestMetadefLoaderPlugin(test_utils.BaseTestCase):
+    @testtools.skipIf(not elasticsearch, "elasticsearch not installed")
     def setUp(self):
         super(TestMetadefLoaderPlugin, self).setUp()
         self.db = unit_test_utils.FakeDB()
